sphinx_search/static/js/rtd_sphinx_search.js

```diff
@@ -788,6 +788,14 @@ window.addEventListener("DOMContentLoaded", () => {
                     form.submit();
                 }
             }
+
+            // close the search modal if the user pressed
+            // Escape button
+            if (e.keyCode === 27) {
+                // Avoid triggering Sphinx's 'hideSearchWords' shortcut.
+                e.stopPropagation();
+                removeSearchModal();
+            }
```

Comment:
but now the modal isn't closed if the panel lost the focus (and hideSearchWords is triggered :/), was able to reproduce by clicking on the "white area" of the modal, but clicking on the search box gains the focus and escape works again.

Reply:
It looks like you'll need to add an event listener for something other than the input box. This key code event is only on the `search_outer_input` element, where you likely also want to listen on `search_outer`.